gha-cache/src/api.rs

```diff
@@ -53,6 +54,9 @@ pub enum Error {
     #[error("Failed to initialize the client: {0}")]
     InitError(Box<dyn std::error::Error + Send + Sync>),
 
+    #[error("Circuit breaker tripped.")]
```

Comment:
Can we make this message more useful to the end user?

Like: "Encountered a 429 Too Many Requests from GitHub Actions Cache, not trying to upload/download from it again"

Reply:
Done!